### PR TITLE
Rotation picture in gallery.

### PR DESCRIPTION
**Pull request by miltonarce**
Rotate Picture 360°

Fixes: *no issue*

Licence: MIT or AGPL

### Description

In this branch, the functionality to rotate images of the gallery was included. The rotate.js Framework was used

### Features

* ...
Do you can rotate the 360 ° images inside the slideshow.
* ...

### Screenshots or screencasts
captura

### Caveats

* ...
* ...

### Tested on

- [x] Windows/Firefox
- [ ] Windows/Chrome
- [ ] Windows/Iexplorer

### TODO

- [ ] ...
- [ ] ...

### Check list

- [ ] Code is properly documented
- [ ] Code is properly formatted
- [ ] Commits have been squashed
- [ ] Tests (unit, integration, api and/or acceptance) are included
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: appinfo/signature.json [hunk 1]
```diff
@@ -0,0 +1,311 @@
+{
```
Comment (DeepDiver1975) [inline appinfo/signature.json]: never add signature.json to git - please remove - THX

File: js/jquery.js [hunk 2]
```diff
@@ -0,0 +1,10364 @@
+/*!
+ * jQuery JavaScript Library v3.3.1
```
Comment (DeepDiver1975) [inline js/jquery.js]: use jquery as provided by owncloud core - please remove

File: l10n/af_ZA.php [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?php
```
Comment (DeepDiver1975) [inline l10n/af_ZA.php]: never manually change translation files - please remove all changes to l10n folder - THX

File: service/filesservice.php [hunk 4]
```diff
@@ -143,13 +142,9 @@ protected function getFolderData($node) {
...
 		if ($nodeType === 'dir') {
 			/** @var Folder $node */
-			try {
- 				if (!$node->nodeExists($this->ignoreAlbum)) {
- 					return [$node];
- 				}
-			} catch (StorageNotAvailableException $e) {
- 				return [];
- 			}
+			if (!$node->nodeExists($this->ignoreAlbum)) {
```
Comment (DeepDiver1975) [inline service/filesservice.php]: why this change? please explain - THX

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

File: appinfo/signature.json (hunk 1 above)
Comment (miltonarce) [reply appinfo/signature.json]: Understood, I removed it.

File: js/jquery.js (hunk 2 above)
Comment (miltonarce) [reply js/jquery.js]: Understood, I removed it

File: l10n/af_ZA.php (hunk 3 above)
Comment (miltonarce) [reply l10n/af_ZA.php]: Sorry, but i didn't change this. Anyway i updated taking files their gallery master.

File: service/filesservice.php (hunk 4 above)
Comment (miltonarce) [reply service/filesservice.php]: Sorry, but i diddn't change this. Anyway i updated taking files their gallery master. Same latest commit.

**Comment by DeepDiver1975** [comment]
looks better now - THX - will have a second look these days

**Comment by miltonarce** [comment]
Sorry, but do i need to do something else?

**Comment by DeepDiver1975** [comment]
> Sorry, but do i need to do something else?

no - I'll give it a look today ....

File: appinfo/app.php [hunk 5]
```diff
@@ -78,6 +78,8 @@ function () use ($c, $appName) {
 		Util::addScript($appName, 'slideshowcontrols');
 		Util::addScript($appName, 'slideshowzoomablepreview');
 		Util::addScript($appName, 'gallerybutton');
+		Util::addScript($appName, 'rotate');
+		Util::addScript($appName, 'rotacion_cup');
```
Comment (phil-davis) [inline appinfo/app.php:82]: I guess this name ``rotacion`` is the Spanish for the English ``rotation``. It makes no difference to the app working - all uses of ``rotacion_cup`` in this PR are consistent.

Comment (miltonarce) [reply appinfo/app.php:82]: Yes, sure.
